components/cam/src/physics/rrtmgp/radiation_utils.F90

```diff
@@ -333,7 +334,9 @@ subroutine handle_error(error_message, fatal)
          if (fatal_local) then
             call endrun(trim(error_message))
          else
-            print *, trim(error_message)
+            if (present(warn)) then
```

Comment:
Did you test this? The way you have this setup, it appears as though you will always throw a warning message so long as ANY value of warn is passed to `handle_error`. I think you should create a `warn_local` variable, set it to `.true.`, and then if `warn` is present then set `warn_local = warn` and check the logical value of `warn_local`. The way it is now you never check the value of `warn`, only if it's present. I.e., you want something like the following:
```
warn_local = .true.
if (present(warn)) warn_local = warn
```
and then within the error checking block:
```
if (fatal_local) then
   call endrun(trim(error_message))
else if (warn_local) then
   print *, trim(error_message)
end if

Reply:
shoot, I actually didn't test it. I'll fix this.